applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.h

```diff
@@ -21,8 +24,78 @@
 #include "includes/element.h"
 
 namespace Kratos {
+/**
+ * @namespace ElementUtilities
+ * @ingroup StructuralMechanicsApplication
+ * @brief This class includes several utilities necessaries for the computation of the different elements
+ * @author Philipp Bucher
+ * @author Vicente Mataix Ferrandiz
+ * @author Riccardo Rossi
+ * @author Ruben Zorrilla
+ */
 namespace StructuralMechanicsElementUtilities {
 
+/// The size type definition
+typedef std::size_t SizeType;
+
+/// The index type definition
+typedef std::size_t IndexType;
+
+/// Node type definition
+typedef Node<3> NodeType;
+
+/// Geometry definitions
+typedef Geometry<NodeType> GeometryType;
+
+/// The zero tolerance
+static constexpr double tolerance = std::numeric_limits<double>::epsilon();
+
+/**
+ * @brief This method performs commons checks on the elements
+ * @param pElement Pointer to the element
+ * @param rCurrentProcessInfo The current process info instance
+ */
+int BaseElementCheck(
+    const Element* pElement,
+    const ProcessInfo& rCurrentProcessInfo
+    );
+
+/**
+ * @brief This method computes the deformation gradient F (for small deformation solid elements)
+ * @param pElement Pointer to the element
+ * @param rF The deformation gradient F
+ * @param rStrainTensor The strain tensor
+ */
+void ComputeEquivalentF(
```

Comment:
These Functions should be templated in the MatrixType.

also we should pass a reference to the element not a constant pointer

Reply:
I am reverting changes right now